### ISSUE TITLE
Component Implementation - VaButton

### ISSUE THREAD

**Issue by timwright12** [department-of-veterans-affairs/va-mobile-app#6870]
## Description

This is a ticket to extract the VA Button component from the mobile app and install it into the component library.

## Important links

- [Current documentation](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/Flagship%20design%20library/Components/Buttons%20and%20Links/VAButton)
- [Current primary code location](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/components/VAButton.tsx)
- [Current primary test location](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/components/VAButton.test.tsx)
- [npm package](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library)
- [Component List](https://docs.google.com/spreadsheets/d/1_EAH2LWSzwF8Om7o4LAYJf6gT9UWENANYpF7SZy3j8w/edit#gid=106854297)

## Definition of Done

- [ ] Component has been added and merged into `main`
- [x] Unit tests are associated with the new component
- [ ] Component has been manually tested inside the app in at least one place
- [ ] New Storybook stories published and working as expected
- [x] Associated documentation added to the docs site
- [ ] A ticket has been created (and communicated to product) for the app teams to install the new component

**Comment by narin** [comment]
@jessicawoodin I'm going to reach out to you about the button colors and how we should be tokenizing them. Jotting some notes here for reference:

## Notes on color tokens
### Found in VADS tokens but weirdly named
  - #B50909 – light mode destructive red (v3-color-error-dark)
  ```json
  "v3-color": {
    "error": {
      "dark": { "value": "#b50909"}
    },
    "base": {
      "darkest": { "value": "#1B1B1B"}
    }
  }
  ```
  - #E1F3F8 – dark mode primary bg pressed (primary-alt-lightest)
  ```json
  "primary": {
      "*": { "value": "#0071bb"},
      "darker": { "value": "#003e73"},
      "darkest": { "value": "#112e51"},
      "alt": {
        "*": { "value": "#02bfe7"},
        "dark": { "value": "#00a6d2"},
        "darkest": { "value": "#046b99"},
        "light": { "value": "#9bdaf1"},
        "lightest": { "value": "#e1f3f8"}
      }
    },
  
  ```
  - #F9DEDE – dark mode destructive pressed (secondary-lightest)
  ```json
  "secondary": {
      "*": { "value": "#e31c3d"},
      "lightest": { "value": "#f9dede"}
    },
  
  ```
### Found in CSS but not color tokens
  - #005EA2 – secondary border color
### Missing entirely
  - Light mode
    - #162E51 – secondary border pressed
    - #5C1111 – light mode destructive pressed
  - Dark mode
    - #58B4FF – primary bg color
    - #FB5A47 – destructive
    - #005EA2 – secondary border pressed

**Comment by narin** [comment]
Tested a single instance within the app and published v0.2.0. Closing out.
